### ISSUE TITLE
Notification when process completes

### ISSUE THREAD

**Issue by iBelieve** [papyros/terminal-app#4]
See the first point in http://www.slant.co/topics/794/viewpoints/27/~linux-terminal-emulator~pantheon-terminal.

**Comment by IdrissDimson** [comment]
I always said the pantheon terminal is how a terminal should be done.

**Comment by zigapk** [comment]
Okay, seems quite a good idea. Should it only display notification when session has no focus to avoid notifications for commands like ls, cd, etc.?

**Comment by iBelieve** [comment]
@zigapk Good point, yes, you wouldn't want notifications when the app is focused.

**Comment by severen** [comment]
I believe a similar feature also exists in Konsole, except it detects when output on the terminal stops (or starts). Correct me if I am wrong, I've been using Gnome for the past six months.

**Comment by gnunn1** [comment]
As an FYI, there is also a Fedora patched version of the GTK VTE and gnome-terminal that supports notifications. The Fedora patch does it via escape codes whereas pantheon terminal does it via DBus. Both are modifying PROMPT_COMMAND to make the call back. You can have a look at the Fedora upstream issue on this for lot's of interesting info and detail this in terms of implementation: 

https://bugzilla.gnome.org/show_bug.cgi?id=711059
